as/cNFT/assembly/market.ts

```diff
@@ -50,6 +50,10 @@ export function set_bid(tokenId: string, bid: Bid): Bid {
     assert_eq_attached_deposit(bid.amount)
     assert_token_exists(tokenId)
 
+    let token = persistent_tokens.get(tokenId);
+
+    assert(token.owner_id != bid.bidder, "You can't bid on your own tokens")
```

Comment:
I think you are also missing another important assert here.
That `bid.bidder == context.predecessor` cause otherwise someone can place bids for other accounts.

Reply:
this also should close this PR: https://github.com/curaOS/contracts/pull/108

Reply:
Yeah. true 👍